GVRf/Framework/jni/objects/vertex_bone_data.cpp

```diff
@@ -50,8 +50,15 @@ int VertexBoneData::getFreeBoneSlot(int vertexId) {
 }
 
 void VertexBoneData::setVertexBoneWeight(int vertexId, int boneSlot, int boneId, float boneWeight) {
-    boneData[vertexId].ids[boneSlot] = boneId;
-    boneData[vertexId].weights[boneSlot] = boneWeight;
+    if (BONES_PER_VERTEX <= boneSlot || 0 > boneSlot) {
+        FAIL("index out of bounds; boneSlot: %d", boneSlot);
+    }
+    if (MAX_BONES <= boneId || 0 > boneId) {
+        FAIL("index out of bounds; boneId: %d", boneId);
+    }
+    BoneData& boneDataElement = boneData.at(vertexId);  //"at" performs bounds-checking
```

Comment:
use of 'at' could throw an exception, which is probably not what we want. and it is proven to be 5x slower than [] operator. (http://stackoverflow.com/questions/9376049/vectorat-vs-vectoroperator) 

I would think it's better to use an "unlikely(..)" compiler directive to check for the bound as other things.